wire-ios-data-model/Source/Model/Message/FileAssetCache.swift

```diff
@@ -975,11 +966,21 @@ private struct FileCache: Cache {
         return cacheFolderURL.appendingPathComponent(safeKey)
     }
 
-    /// Deletes all existing caches. After calling this method, existing caches should not be used anymore.
-    /// This is intended for testing
+    /// Deletes the contents of the cache.
+
     func wipeCaches() {
         logger.debug("wiping cache")
-        _ = try? FileManager.default.removeItem(at: cacheFolderURL)
+        if FileManager.default.fileExists(atPath: cacheFolderURL.path) {
+            do {
+                // Delete the entire cache.
+                try FileManager.default.removeItem(at: cacheFolderURL)
+            } catch {
+                logger.error("failed to wipe cache: \(error)")
+            }
+        }
+
+        // Create it again so we can write files to it.
+        FileManager.default.createAndProtectDirectory(at: cacheFolderURL)
```

Comment:
in case the remove the cache folder fail, would that method fail too?

Reply:
I'm not sure, but if it does it may crash since there's a fatal inside. I'll make a throwing version and throw the errors up

Reply:
done